Model/Sync/Catalog/Processor.php

```diff
@@ -355,13 +355,15 @@ public function syncItems($collectionItems, $storeId, $visibleVariants = false)
                 if (!$visibleVariants) {
                     $parentData = $this->pushParentData((int)$itemId, $tempSqlArray, $parentData, $parentIds);
                 }
-                $syncDataSql = [];
-                $syncDataSql[] = $tempSqlArray;
-                $this->insertOnDuplicate(
-                    'yotpo_product_sync',
-                    $syncDataSql
-                );
-                $sqlData[] = $tempSqlArray;
+                if ($tempSqlArray) {
```

Comment:
@manaftryzens can you please explain this fix?
What was the bug?

Reply:
We need to update the sync table only if the $tempSqlArray has some entries. Otherwise it will insert the row with yotpo_id as 0 for that product, which is unwanted.

Reply:
@manaftryzens I'm afraid that this fix won't work then.
So `$tempSqlArray` is an actual map, not an array. Please rename to `$tempSqlMap`.

The variable is initiated on L286 `$tempSqlArray = [` so it will always pass the condition.
Later it is being processed on `$returnResponse = $this->processResponse(`, and returned as the `temp_sql` key of the processed response.

The condition will always return true.
The if-statement must be modified to check the Map in the relevant Yotpo id key.